code/modules/martial_arts/tae_clown_do.dm

```diff
@@ -0,0 +1,99 @@
+//HONK DUB-DUB WAJUB HONK!
+/datum/martial_art/tae_clown_do
+	name = "Tae Clown Do"
+	has_explaination_verb = TRUE
+	combos = list(/datum/martial_combo/tae_clown_do/banana_pie_palm, /datum/martial_combo/tae_clown_do/clown_PUNch, /datum/martial_combo/tae_clown_do/disclownbobulate)
+
+/datum/martial_art/tae_clown_do/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	MARTIAL_ARTS_ACT_CHECK
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	var/atk_verb = pick("slaps", "honkstomps", "wallops", "thwacks", "bifs")
+	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>", \
+					  "<span class='userdanger'>[A] [atk_verb] you!</span>")
+	D.apply_damage(13, BRUTE)
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, 1, -1)
+	if(prob(50))
+		A.say(pick("POW!", "THUD!", "WHAP!", "ZUNK!", "BOOM!", "BLAP!", "KAPOW!"))
+	if(prob(D.getBruteLoss()))
+		D.visible_message("<span class='warning'>[D] stumbles and falls!</span>", "<span class='userdanger'>The blow sends you to the ground!</span>")
+		D.Weaken(3)
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Punched", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/tae_clown_do/explaination_header(user)
+	to_chat(usr, "<b><i> You recall the teachings of Clown Honk Hi, internalizing them and becoming a master of Tae Clown Do.</i></b>")
+
+/datum/martial_art/tae_clown_do/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	MARTIAL_ARTS_ACT_CHECK
+	var/obj/item/grab/G = D.grabbedby(A,1)
+	if(G)
+		if(istype(D, /mob/living/carbon/human))
+			if(isobj(D.shoes))
+				D.unEquip(D.shoes) // Removes shoes
```

Comment:
```suggestion
	if(G && ishuman(D) && D.shoes)
		D.unEquip(D.shoes) // Removes shoes
```
best to write it like this. `ishuman` is preferred. the `shoes` variable can only ever be a value of type `item`. So just a normal check works there